node/pkg/common/readiness.go

```diff
@@ -1,33 +1,41 @@
 package common
 
-import "github.com/certusone/wormhole/node/pkg/readiness"
+import (
+	"fmt"
+
+	"github.com/certusone/wormhole/node/pkg/readiness"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+)
 
 const (
-	ReadinessEthSyncing       readiness.Component = "ethSyncing"
-	ReadinessSolanaSyncing    readiness.Component = "solanaSyncing"
-	ReadinessTerraSyncing     readiness.Component = "terraSyncing"
-	ReadinessAlgorandSyncing  readiness.Component = "algorandSyncing"
-	ReadinessNearSyncing      readiness.Component = "nearSyncing"
-	ReadinessAptosSyncing     readiness.Component = "aptosSyncing"
-	ReadinessSuiSyncing       readiness.Component = "suiSyncing"
-	ReadinessBSCSyncing       readiness.Component = "bscSyncing"
-	ReadinessPolygonSyncing   readiness.Component = "polygonSyncing"
-	ReadinessAvalancheSyncing readiness.Component = "avalancheSyncing"
-	ReadinessOasisSyncing     readiness.Component = "oasisSyncing"
-	ReadinessAuroraSyncing    readiness.Component = "auroraSyncing"
-	ReadinessFantomSyncing    readiness.Component = "fantomSyncing"
-	ReadinessKaruraSyncing    readiness.Component = "karuraSyncing"
-	ReadinessAcalaSyncing     readiness.Component = "acalaSyncing"
-	ReadinessKlaytnSyncing    readiness.Component = "klaytnSyncing"
-	ReadinessCeloSyncing      readiness.Component = "celoSyncing"
-	ReadinessMoonbeamSyncing  readiness.Component = "moonbeamSyncing"
-	ReadinessNeonSyncing      readiness.Component = "neonSyncing"
-	ReadinessTerra2Syncing    readiness.Component = "terra2Syncing"
-	ReadinessInjectiveSyncing readiness.Component = "injectiveSyncing"
-	ReadinessXplaSyncing      readiness.Component = "xplaSyncing"
-	ReadinessPythNetSyncing   readiness.Component = "pythnetSyncing"
-	ReadinessArbitrumSyncing  readiness.Component = "arbitrumSyncing"
-	ReadinessOptimismSyncing  readiness.Component = "optimismSyncing"
-	ReadinessBaseSyncing      readiness.Component = "baseSyncing"
-	ReadinessWormchainSyncing readiness.Component = "wormchainSyncing"
+	ReadinessEthSyncing readiness.Component = "ethSyncing"
 )
+
+// RegisterReadinessSyncing registers the specified chain for readiness syncing. It will panic if the chain ID is invalid
+// so it should only be used during initialization.
+func RegisterReadinessSyncing(chainID vaa.ChainID) {
```

Comment:
See comment on review, we could do this automatically to make usage of the API more intuitive.

Reply:
I'm not crazy about auto registering (especially since `readiness.RegisterComponent` currently panics in that case.

I made `MustRegisterReadinessSyncing` both register and return the tag, and am calling it from watcher start up, so there is no redundancy.